content/6_Advanced/Multiplicative.mdx

```diff
@@ -1,24 +1,230 @@
+	}
+	i--;  // Decrementing i to match the last value used in the loop
+	if (i == n / i) {
+		sum -= i * (n / i);  // Adjusting the sum if n is a perfect square
+	}
+	cout << sum << endl;
+}
+```
+
+</CPPSection>
+</LanguageSection>
+
+## Example - Totient Sum
+
+<FocusProblem problem="f2" />
+
+### Explanation
+
+Currently, the formulas related to Euler's totient function mentioned in this
+article are limited. Can we use them to simplify the problem? The answer is yes,
+and now we'll utilize the formula $\sum_{d|n}\varphi(d)=n$ to simplify the
+expression.
+
+This formula can also be seen as $\varphi(n)=n-\sum_{d|n, d<n}\varphi(d)$. Let's
+denote $\phi(n)=\sum_{i=1}^n \varphi(i)$, then we have:
+
+$$
+\phi(n)=\sum_{i=1}^n\varphi(i)=\sum_{i=1}^n i−\sum_{d|i,d<i}\varphi(d)=\frac{n\cdot(n+1)}{2}−\sum_{i=2}^n\sum_{d|i,d<i}\varphi(d)=\frac{n\cdot(n+1)}{2}−\sum_{i=2}^{n}\sum_{d=1}^{\left\lfloor\frac{n}{\left(\frac{i}{d}\right)}\right\rfloor}\varphi(d)
+$$
+
+$$
+n \cdot \frac{(n+1)}{2} - \sum_{i=2}^{n} \sum_{d=1}^{\lfloor \frac{n}{i} \rfloor} \phi(d) = n \cdot \frac{(n+1)}{2} - \sum_{i=2}^{n} \phi(\lfloor \frac{n}{i} \rfloor)
+$$
+
+So as long as you calculate the values of $\phi(\lfloor \frac{n}{i} \rfloor$ for
+$O(\sqrt{n})$ times, you can compute $\phi(n)$. What about the complexity of
+such an operation?
+
+Suppose the complexity of calculating $\phi(n)$ is $T(n)$, then we have
+$T(n) = O(\sqrt{n}) + \sum_{i=1}^{\sqrt{n}} T(i) + T(n/i)$. **Here, we only
+expand one layer because deeper complexities are higher-order terms**. So, we
+have
+$T(n) = \sum_{i=1}^{\sqrt{n}} O(\sqrt{i}) + O(\sqrt{\frac{n}{i}}) = O(n^{3/4})$.
+
+Since $\phi(n)$ is a prefix sum of a multiplicative function, the sieve method
+can preprocess a portion. Assuming the preprocessing includes the first $k$
+positive integers' $\phi(n)$ where $k \geq \sqrt{n}$, the complexity becomes
+$T(n) = \sum_{i=1}^{k} {\sqrt{\frac{n}{i}}} = O({\frac{n}{\sqrt{k}}})$. When
+$k = O(n^{2/3})$, we can achieve a good complexity of $T(n) = O(n^{2/3})$.
+
+How did we come up with the place where we utilized
+$\varphi(n) = n-\sum_{d|n,d<n} \varphi(d)$? Let's take a look at this:
+
+$$
+\frac{n \cdot (n+1)}{2} = \sum_{i=1}^{n} i = \sum_{i=1}^{n} \sum_{d} [d|i]\varphi(d) = \sum_{i=2}^{n} \sum_{d=1}^{\left\lfloor \frac{n}{\left(\frac{i}{d}\right)} \right\rfloor} \varphi(d) = \sum_{i=1}^{n} \phi(\left\lfloor \frac{n}{i} \right\rfloor)
+$$
+
+**If we can construct a function through Dirichlet convolution that computes
+prefix sums more efficiently and if another function suitable for convolution is
+also easy to calculate, we can simplify the calculation process.** For example,
+in the above problem, we used the property of $\varphi * I = id$. But remember,
+not all problems of this type can be easily solved by just pairing with an
+identity function $I$. Sometimes, a more careful observation is needed.
+
+### Implementation
+
+**Time Complexity: $\mathcal{O}(n^{3/4})$**
+
+<LanguageSection>
+<CPPSection>
```

Comment:
do you think you can fix up this implementation
i find it clashes with the rest of the module